gnosis/test/api-test.ini

```diff
@@ -162,9 +162,47 @@ proxy_storage:
     andBool #rangeBytes( 32,        SAFE_TX_HASH)
     andBool #rangeBytes( 32,        DOMAIN_SEPARATOR)
 
+[testSignatureSplit]
+statusCode: _ => EVMC_SUCCESS
+this: PROXY_ID
+msg_sender: MSG_SENDER
+callData: #abiCallData("testSignatureSplit", #bytes(#encSig(SIG0_R, SIG0_S, SIG0_V) ++ #encSig(SIG1_R, SIG1_S, SIG1_V) ++ #encSig(SIG2_R, SIG2_S, SIG2_V)), #uint256(POS))
+callValue: 0
+proxy_storage:
+    _:Map
++requires:
+    // Range
+    andBool #rangeAddress(PROXY_ID)
+    andBool #rangeAddress(MSG_SENDER)
+    andBool #rangeUInt(256, SIG0_R)
+    andBool #rangeUInt(256, SIG0_S)
+    andBool #rangeUInt(8  , SIG0_V)
+    andBool #rangeUInt(256, SIG1_R)
+    andBool #rangeUInt(256, SIG1_S)
+    andBool #rangeUInt(8  , SIG1_V)
+    andBool #rangeUInt(256, SIG2_R)
+    andBool #rangeUInt(256, SIG2_S)
+    andBool #rangeUInt(8  , SIG2_V)
+    andBool #rangeUInt(256, POS)
+
+[testSignatureSplit-pos0]
+output: _ => #encodeArgs(#uint8(SIG0_V), #bytes32(SIG0_R), #bytes32(SIG0_S))
```

Comment:
The construct `#encodeSig` leads to complications here. As you see, output has to be specified separately for each pos value. A more general approach would be to specify input as #toBytes(DATA, 65*N), and output as a certain bytes range depending on `pos`. Then you could move specification for output to the main spec.

You might need an additional construct for a byte range, like `#byteRange(DATA, START, LEN, N)`.

In addition I'm wondering if it's possible to prove the spec for all pos values <=N, but we can leave this for later.

Reply:
#toBytes(DATA, 65*N) may cause some additional problems for writing specs and symbolic reasoning. First, current java backend doesn't support  looking up symbolic keys from map. we need to enumerate all the possible pos anyway. Second, If we use #toBytes(DATA, 65*N), when loading value from specific memory location, the value will be like #asWord(....) instead of a single value like `SIG0_R`and the output will be very large. If I want to further constraint the SIG0_V, SIG0_R and SIG0_S, instead of directly writing constraint on them, I need to write constraint on three byte arrays.

Reply:
That's why I proposed to have a byte range construct. Everywhere you have SIG0_V, SIG0_R and SIG0_S now, you would have a certain byte range over DATA instead. And you can constrain it the same way as we constrain data. The advantage is that you can write a single spec, with output fully specified, for ANY N and POS, and restrict LEN and POS in sub-specs byt additional +requires clauses. I guess it would give us more flexibility on the long term in other areas too.

The spec will remain compact. And later we can think of some abstraction that avoids unfolding the byte range into nthbyteof sequences, like the one Daejun did for hash1 and hash2. Then configuration will stay compact too.

Whenever we have repetition this is a bad sign.

For `#asWord` construct that appears when reading from memory, you might need some additional lemmas like the lemmas that I added at the end of verification.k.

In principle I can do this refactoring, but then it will interfere with your work because you'll have to merge your current changes with refactored spec.

Reply:
I think you can first go ahead and refactor it as I'm working on other stuff now. @daejunpark Could you share your opinion before the refactoring starts?

Reply:
Yes, @daejunpark we need your opinion on this. Do you prefer Yi's implementation or I should refactor according to my proposal?

I'll merge in the meantime. The refactoring will be a separate PR.